### ISSUE TITLE
smartcn_tokenizer tokenizes all punctuation into a comma

### ISSUE THREAD

**Issue by pkphlam** [elastic/elasticsearch#21418]
**Elasticsearch version**: 5.0

**Plugins installed**: smartcn

Not sure if anybody else can reproduce this, but I'm finding that the smartcn_tokenizer is treating all punctuation as a separate comma token instead of ignoring punctuation as is typical with the other tokenizers.

For example, running the string `,!.?@#` through the Analysis API on a text field analyzed with the smartcn_tokenizer gives the result:

```
{'tokens': [{'end_offset': 1,
   'position': 0,
   'start_offset': 0,
   'token': ',',
   'type': 'word'},
  {'end_offset': 2,
   'position': 1,
   'start_offset': 1,
   'token': ',',
   'type': 'word'},
  {'end_offset': 3,
   'position': 2,
   'start_offset': 2,
   'token': ',',
   'type': 'word'},
  {'end_offset': 4,
   'position': 3,
   'start_offset': 3,
   'token': ',',
   'type': 'word'},
  {'end_offset': 5,
   'position': 4,
   'start_offset': 4,
   'token': ',',
   'type': 'word'},
  {'end_offset': 6,
   'position': 5,
   'start_offset': 5,
   'token': ',',
   'type': 'word'}]}
``` 

Doing the same on an identically mapped index with either the standard or icu tokenizers does not have this problem. Punctuation is correctly ignored with those two tokenizers.

**Comment by clintongormley** [comment]
Hi @pkphlam 

Yeah, the smartcn analyzer adds a stopwords token filter which basically removes all punctuation.  The tokenizer is just a tokenizer, so it leaves punctuation in place.  We should probably expose the smartcn stop words either via the `stop` filter, or specifically in the `smartcn` plugin.

**Comment by pkphlam** [comment]
Thanks for the info. It seems a little bit unusual that the tokenizer itself retains the punctuation. As far as I can tell, all the other tokenizers that do similar things (standard, icu, etc.) seem to omit punctuation by default. So currently, if I wanted to use the smartcn tokenizer in a custom analyzer and remove punctuation, I'd have to add a custom character filter.

**Comment by clintongormley** [comment]
@pkphlam or you can do what the smartcn analyzer itself does (https://github.com/apache/lucene-solr/blob/master/lucene/analysis/smartcn/src/java/org/apache/lucene/analysis/cn/smart/SmartChineseAnalyzer.java#L103) and use  stopwords file like the following: https://github.com/apache/lucene-solr/blob/master/lucene/analysis/smartcn/src/resources/org/apache/lucene/analysis/cn/smart/stopwords.txt

**Comment by clintongormley** [comment]
Discussed in FixItFriday - let's expose the `_smartcn_` stopwords in the stop analyzer.

**Comment by romseygeek** [comment]
cc @elastic/es-search-aggs

**Comment by Razi007** [comment]
Is issue is still active? Intrested to work here.

**Comment by mayya-sharipova** [comment]
@Razi007  Yes, the issue is still active, and since there is no PR referenced here, nobody is working on it. Please go ahead, your contribution is very appreciated.

**Comment by Razi007** [comment]
@mayya-sharipova for mentorship should we discuss here or on free-node.Also please guide me which part of code should I look for better understanding.

**Comment by mayya-sharipova** [comment]
@Razi007 sorry for a late reply.  Let me discuss the implementation details with my team

**Comment by mayya-sharipova** [comment]
> Discussed in FixItFriday - let's expose the `_smartcn_` stopwords in the stop analyzer.

This will introduce dependency on `org.apache.lucene:lucene-analyzers-smartcn` in the server code. I am not sure if we want to have this dependency.

**Comment by mayya-sharipova** [comment]
We have discussed it on Fixit Friday, and it first requires investigation why smartcn_tokenizer doesn't use punctuation marks for tokenization.

@Razi007 Sorry, I don't  think this issue is a good as a fist issue, as it requires some investigation.
